Source/WebCore/page/SecurityOrigin.cpp

```diff
@@ -440,6 +440,28 @@ bool SecurityOrigin::isSameOriginAs(const SecurityOrigin& other) const
     return isSameSchemeHostPort(other);
 }
 
+bool SecurityOrigin::isSameSiteAs(const SecurityOrigin& other) const
+{
+#if ENABLE(PUBLIC_SUFFIX_LIST)
+    // https://html.spec.whatwg.org/#sites:same-site
+    if (isOpaque() != other.isOpaque())
+        return false;
+    if (!isOpaque() && protocol() != other.protocol())
+        return false;
+
+    if (isOpaque())
+        return isSameOriginAs(other);
+
+    auto topDomain = topPrivatelyControlledDomain(domain());
+    if (topDomain.isEmpty())
+        return host() == other.host();
+
+    return topDomain == topPrivatelyControlledDomain(other.domain());
+#else
+    return false;
+#endif // ENABLE(PUBLIC_SUFFIX_LIST)
```

Comment:
Doesn't this mean that if you don't have PUBLIC_SUFFX_LIST enabled this method doesn't return anything? Shouldn't it fallback to same origin?

Not sure it really makes sense to have PUBLIC_SUFFIX_LIST as optional. I'm somewhat curious if there was prior discussion about removing it.

(Also somewhat surprised there's no prior code covering this check.)

Reply:
Sure I could have it fallback.

I don't think there is a problem with our ports requiring PUBLIC_SUFFIX_LIST.